src/shared/cluster_utils.c

```diff
@@ -59,6 +59,61 @@ int w_is_worker(void) {
     return is_worker;
 }
 
+int w_is_single_node(int* is_worker) {
+    OS_XML xml;
+    const char * xmlf[] = {"ossec_config", "cluster", NULL};
+    const char * xmlf2[] = {"ossec_config", "cluster", "node_type", NULL};
+    const char * xmlf3[] = {"ossec_config", "cluster", "disabled", NULL};
+    const char *cfgfile = OSSECCONF;
+    int _is_worker = OS_INVALID;
+    int is_single_node = OS_INVALID;
+
+    if (OS_ReadXML(cfgfile, &xml) < 0) {
+        mdebug1(XML_ERROR, cfgfile, xml.err, xml.err_line);
+    } else {
+        char * cl_config = OS_GetOneContentforElement(&xml, xmlf);
+        if (cl_config && cl_config[0] != '\0') {
+            char * cl_type = OS_GetOneContentforElement(&xml, xmlf2);
+            if (cl_type && cl_type[0] != '\0') {
+                char * cl_status = OS_GetOneContentforElement(&xml, xmlf3);
+                if(cl_status && cl_status[0] != '\0'){
+                    if (!strcmp(cl_status, "no")) {
+                        is_single_node = 0;
+                        if (!strcmp(cl_type, "client") || !strcmp(cl_type, "worker")) {
+                            _is_worker = 1;
+                        } else {
+                            _is_worker = 0;
+                        }
+                    } else {
+                        is_single_node = 1;
+                    }
+                } else {
+                    is_single_node = 0;
```

Comment:
The cluster is able to work without the **disabled** tag. This behavior should be described in the documentation because as you say, it isn't consistent. We'll talk with the corresponding team.

Reply:
Fixed!